sdk/proto/analyzer.proto

```diff
@@ -39,20 +39,58 @@ service Analyzer {
-    string name = 4;
+    string type = 1;                                                    // the type token of the resource.
+    google.protobuf.Struct properties = 2;                              // the full properties to use for validation.
+    string urn = 3;                                                     // the URN of the resource.
+    string name = 4;                                                    // the name for the resource's URN.
+    AnalyzerResourceOptions options = 5;                                // the resource options.
+    AnalyzerProviderResource provider = 6;                              // the resource's provider.
 }
 
-// Resource defines the view of a Pulumi-managed resource as sent to Analyzers. The properties
+// AnalyzerResource defines the view of a Pulumi-managed resource as sent to Analyzers. The properties
 // of the resource are specific to the type of analysis being performed. See the Analyzer
 // service definition for more information.
 message AnalyzerResource {
+    string type = 1;                                                    // the type token of the resource.
+    google.protobuf.Struct properties = 2;                              // the full properties to use for validation.
+    string urn = 3;                                                     // the URN of the resource.
+    string name = 4;                                                    // the name for the resource's URN.
+    AnalyzerResourceOptions options = 5;                                // the resource options.
+    AnalyzerProviderResource provider = 6;                              // the resource's provider.
+    string parent = 7;                                                  // an optional parent URN that this child resource belongs to.
+    repeated string dependencies = 8;                                   // a list of URNs that this resource depends on.
+    map<string, AnalyzerPropertyDependencies> propertyDependencies = 9; // a map from property keys to the dependencies of the property.
+}
+
+// AnalyzerResourceOptions defines the options associated with a resource.
+message AnalyzerResourceOptions {
+    // CustomTimeouts allows a user to be able to create a set of custom timeout parameters.
+    message CustomTimeouts {
+        double create = 1; // The create resource timeout in seconds.
+        double update = 2; // The update resource timeout in seconds.
+        double delete = 3; // The delete resource timeout in seconds.
+    }
+
+    bool protect = 1;                            // true if the resource should be marked protected.
+    repeated string ignoreChanges = 2;           // a list of property names to ignore during changes.
+    bool deleteBeforeReplace = 3;                // true if this resource should be deleted before replacement.
+    bool deleteBeforeReplaceDefined = 4;         // true if the deleteBeforeReplace property should be treated as defined even if it is false.
```

Comment:
This is slightly funny - I guess it's fine - but is this trying to model a tri-state value space for `deleteBeforeReplace`?

Reply:
> is this trying to model a tri-state value space for `deleteBeforeReplace`?

Yes. This is how it is modeled for `RegisterResource`:

https://github.com/pulumi/pulumi/blob/137fd54f1cdd5ff901ddba0f281039bf3b9a5e01/sdk/proto/resource.proto#L92

and

https://github.com/pulumi/pulumi/blob/137fd54f1cdd5ff901ddba0f281039bf3b9a5e01/sdk/proto/resource.proto#L100

However, in that case, it was because `deleteBeforeReplace` was already defined, and we need to add `deleteBeforeReplaceDefined` as a way of explicitly setting false, while remaining compatible with the existing interface. I suppose if we had thought of that from the start, we'd model it as an enum.

So, I could do the cleaner thing now and model this as an enum, if that's preferred. (Though, I could also argue that it's somewhat nice to be consistent with how we do it for `RegisterResource`).

Reply:
Got it - makes sense to stay consistent.